src/validators/validateBin.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Validate the `bin` field in a package.json, which can either be a string
+ * with the path to the executable, or a Record&lt;string, string&gt; where the
+ * key is the name of the command and the value is the path to the executable.
+ *
+ * {
+ *   "my-cli" : "./cli.js",
+ *   "my-dev-tool" : "./dev-tool.js",
+ * }
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const validateBin = (obj: any): string[] => {
```

Comment:
[Types? Bug] Including an `any` is overly permissive when `unknown` will do:

```suggestion
export const validateBin = (obj: unknown): string[] => {
```

Then later on:

```diff
- for (const key in obj) {
+ for (const [key, value] of Object.entries(obj)) {
```